api/src/repositories/project-repository.ts

```diff
@@ -137,13 +137,77 @@ export class ProjectRepository extends BaseRepository {
       sqlStatement.append(SQL`}'`);
     }
 
+    if (pagination?.sort) {
+      sqlStatement
+        .append(SQL`ORDER BY `)
+        .append(pagination.sort)
+        .append(SQL` `);
+
+      if (pagination.order) {
+        sqlStatement.append(pagination.order === 'asc' ? SQL`ASC` : SQL`DESC`);
+      }
+    }
+
+    if (pagination) {
+      sqlStatement.append(SQL`
+        LIMIT
+          ${pagination.limit}
+        OFFSET
+          ${(pagination.page - 1) * pagination.limit}
+      `);
+    }
+
     sqlStatement.append(';');
 
     const response = await this.connection.sql(sqlStatement, ProjectListData);
 
     return response.rows;
   }
 
+  /**
+   * Returns the total count of projects that are visible to the given user.
+   *
+   * @param {boolean} isUserAdmin
+   * @param {(number | null)} systemUserId
+   * @return {*}  {Promise<number>}
+   * @memberof ProjectRepository
+   */
+  async getProjectCount(isUserAdmin: boolean, systemUserId: number | null): Promise<number> {
+    const sqlStatement = SQL`
+      SELECT
+        COUNT(*) as project_count
+      FROM
+        project as p
+      WHERE 1 = 1
+    `;
+
+    if (!isUserAdmin) {
+      sqlStatement.append(SQL`
+        AND p.project_id IN (
+          SELECT
+            project_id
+          FROM
+            project_participation
+          where
+            system_user_id = ${systemUserId}
+        )
+      `);
+    }
+
+    sqlStatement.append(';');
+
+    const response = await this.connection.sql(sqlStatement, z.object({ project_count: z.string().transform(Number) }));
+
+    if (response?.rowCount < 1) {
+      throw new ApiExecuteSQLError('Failed to get project count', [
+        'ProjectRepository->getProjectCount',
+        'rows was null or undefined, expected rows != null'
+      ]);
+    }
```

Comment:
We probably have to remove this, as a user could legitimately have no projects.

Reply:
This query is supposed to return exactly one row, which captures the project count. So if a user has no projects, the result of this query should be `[{ project_count: 0 }]`, hence the check for `response.rowCount`

Reply:
Ah gotcha, fair